cumulus_library/studies/core/core_templates/documentreference.sql.jinja

```diff
@@ -67,8 +66,14 @@ SELECT DISTINCT
     tdr.author_year,
     tdr.doc_format_code,
     tdr.subject_ref,
-    coalesce(tdr.encounter_ref, context_encounter.encounter.reference) as encounter_ref,
+    {% if schema["documentreference"]["context"]["encounter"] %}
+    context_encounter.encounter.reference AS encounter_ref,
+    {% else %}
+    CAST(NULL AS VARCHAR) AS encounter_ref,
+    {% endif %}
     concat('DocumentReference/', tdr.id) AS doc_ref
-FROM temp_documentreference AS tdr,
-    unnest(context.encounter) AS context_encounter (encounter) --noqa
+FROM temp_documentreference AS tdr
+{% if schema["documentreference"]["context"]["encounter"] %}
+    , unnest(context.encounter) AS context_encounter (encounter) --noqa
+{% endif %}
```

Comment:
Do we want an unnest here? This will create duplicate DocRef rows if there are multiple encounters in that array, yeah?

Reply:
i think this has to be ok so you can traverse backwards, right? generally we're doing distinct ID checks on the primary resource when we're doing thinks like counts

this happens a lot with the core tables, for things like coding systems, for example - you've got to handle this in some way in your study queries.

Reply:
i've got to touch the core tables docs as part of my PR, i'll add a note at least there. I also have a WIP 'notes for study authors' doc this could get into.